### DISCUSSION TITLE
how to send ctrl d with bash?

### DISCUSSION THREAD

**Discussion by lockcher4** [termux/termux-app#3160] [Q&A] [answered]
I downloaded termux:widget to run a script to kill all termux processes, while adding my commands to the bash history.  I made this script 

#!/bin/bash
kill -1 $(ps -A) 

but after itself it leaves unclosed windows( I want it to close

**Comment by agnostic-apollo** [comment, marked as answer]
Run `am startservice -n com.termux/.app.TermuxService -a com.termux.service_stop` if you want to kill all termux sessions and tasks and exit the app.

**Reply by lockcher4** [reply]
after that the history is not saved

**Reply by agnostic-apollo** [reply]
Yeah, that will happen currently, I guess you can manually send kill signals first and then stop the service.
